src/plugins/data_views/common/content_management/types.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  SavedObject,
+  SavedObjectReference,
+  SavedObjectsResolveResponse,
+  SavedObjectsUpdateResponse,
+} from '@kbn/core-saved-objects-api-server';
+import type { SavedObjectsMigrationVersion } from '@kbn/core-saved-objects-common';
+
+import type {
+  GetIn,
+  CreateIn,
+  SearchIn,
+  UpdateIn,
+  DeleteIn,
+} from '@kbn/content-management-plugin/common';
+
+import { DataViewAttributes } from '../types';
+import { DataViewContentType } from './constants';
+
+interface Reference {
+  type: string;
+  id: string;
+}
+
+export type DataViewSavedObject = SavedObject<DataViewAttributes>;
+
+// ----------- GET --------------
+
+export type DataViewGetIn = GetIn<typeof DataViewContentType>;
+
+export interface DataViewGetOut {
+  savedObject: SavedObjectsResolveResponse<DataViewAttributes>['saved_object'];
+  outcome: SavedObjectsResolveResponse['outcome'];
+  aliasTargetId: SavedObjectsResolveResponse['alias_target_id'];
+  aliasPurpose: SavedObjectsResolveResponse['alias_purpose'];
+}
+
+// ----------- CREATE --------------
+
+export interface CreateOptions {
+  /** If a document with the given `id` already exists, overwrite it's contents (default=false). */
+  overwrite?: boolean;
+  migrationVersion?: SavedObjectsMigrationVersion;
+  /** A semver value that is used when upgrading objects between Kibana versions. */
+  coreMigrationVersion?: string;
+  /** Array of referenced saved objects. */
+  references?: SavedObjectReference[];
+}
+
+export type DataViewCreateIn = CreateIn<
+  typeof DataViewContentType,
+  DataViewAttributes,
+  CreateOptions
```

Comment:
You are explicitely declaring that `DataViewCreateIn` takes `CreateOptions`. You now have to pass it in [saved_objects_client_wrapper.ts](https://github.com/elastic/kibana/pull/153733/files#diff-6b27e023bccc10d8ab7af4a5b7c4b39892bb623ac47c2949e8c21e035dcf5e1d) L91